services/worker-manager/src/provisioner.js

```diff
@@ -128,6 +189,21 @@ class Provisioner {
           } catch (err) {
             this.monitor.reportError(err, {providerId: pId}); // Just report this and move on
           }
+
+          if (!providerByPool.providers.has(pId)) {
```

Comment:
This took me a while to figure out, so maybe a comment?
```suggestion
          // If this provider no longer has any workers, ask it to remove its resources
          // and remove it from the previousProviderIds list
          if (!providerByPool.providers.has(pId)) {
```